pkg/resspec/vmspec_matcher.go

```diff
@@ -420,9 +416,37 @@ func GetVMSpec(ctx context.Context, nodeResources *edgeproto.NodeResources, cli
 	if skippedExtraRes > 0 {
 		reasons = append(reasons, fmt.Sprintf("%d with optional resources not requested", skippedExtraRes))
 	}
+	for resName, count := range skippedExtra {
+		reasons = append(reasons, fmt.Sprintf("%d with %s not requested", count, resName))
+	}
 	return &vmspec, errors.New("no suitable infra flavor found for requested node resources, " + strings.Join(reasons, ", "))
 }
 
+// GPU requirements must match exactly to avoid wasting gpu resources
```

Comment:
So in case I don't care what type of nvidia gpu I get as long as it has >8G of vram, there is still a hard requirement of the nvidia gpu model. Long term I see it as a good idea to loosen this requirement for example in case of autoprov deploying apps to different sites. What do you think?

Reply:
I agree, long term this could change into something more flexible. Perhaps specifying a GPU family like "ampere" plus a VRAM amount (assuming GPUs can be scheduled with VRAM limits). Right now though I think we have to stick with the specific model id.